react/Installments.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useContext } from 'react'
+import { defineMessages, FormattedNumber } from 'react-intl'
+import { ProductContext } from 'vtex.product-context'
+import { FormattedCurrency } from 'vtex.format-currency'
+import { useCssHandles } from 'vtex.css-handles'
+import { IOMessageWithMarkers } from 'vtex.native-types'
+
+import { StorefrontFC } from './types'
+
+interface Props {
+  message: string
+  markers: string[]
+}
+
+interface Installment {
+  Value: number
+  InterestRate: number
+  TotalValuePlusInterestRate: number
+  NumberOfInstallments: number
+}
+
+const CSS_HANDLES = [
+  'installments',
+  'installmentsNumber',
+  'installmentValue',
+  'installmentsTotalValue',
+  'interestRate',
+] as const
+
+const Installments: StorefrontFC<Props> = props => {
+  const { message, markers } = props
+  const handles = useCssHandles(CSS_HANDLES)
+  const { selectedItem } = useContext(ProductContext)
+
+  if (!selectedItem) {
+    return null
+  }
+
+  const commercialOffer = selectedItem?.sellers[0]?.commertialOffer
+  if (!commercialOffer) {
+    return null
+  }
+
+  const {
+    Value,
+    InterestRate,
+    TotalValuePlusInterestRate,
+    NumberOfInstallments,
+  } = commercialOffer.Installments.reduce(
+    (previous: Installment, current: Installment) =>
+      previous.NumberOfInstallments > current.NumberOfInstallments
+        ? previous
+        : current
+  )
+
+  return (
+    <div className={handles.installments}>
+      <IOMessageWithMarkers
+        message={message}
+        markers={markers}
+        handleBase="listPrice"
+        values={{
+          installmentsNumber: (
```

Comment:
I think we do need to know if the installments has interest or not otherwise this component would be coupled to the settings of search query and other places